### ISSUE TITLE
Using sqlite extensions with sqlite3 gem

### ISSUE THREAD

**Issue by cboulanger** [sparklemotion/sqlite3-ruby#383]
Hi, I am interested in using the spellfix extension of sqlite.

https://www.sqlite.org/spellfix1.html
https://github.com/sqlite/sqlite/blob/master/ext/misc/spellfix.c

is this possible? Of course, it would be great if it would be possible without compilation (via the OS sqlite?). 

Thank you.

**Comment by cboulanger** [comment]
Hi, thanks for your response! I assume that the native sqlite binary that comes with the default gem or bundle install (I am on Ubuntu) doesn't have that (or any other) extension bundled - so I would need to compile it myself as documented [here](https://www.sqlite.org/loadext.html), I guess.

**Comment by flavorjones** [comment]
@cboulanger Yep, I was able to do this on my linux machine (after cloning sqlite/sqlite)

```sh
# in sqlite directory
gcc -g -fPIC -shared ./ext/misc/spellfix.c -o spellfix.o
```

then, in ruby:

```ruby
#! /usr/bin/env ruby

require "sqlite3"

db = SQLite3::Database.new(':memory:')
db.enable_load_extension(true)
db.load_extension("/path/to/sqlite/spellfix.o")
db.execute("CREATE VIRTUAL TABLE demo USING spellfix1;")
```

I'm going to close this, but please do comment if you have any other questions!

**Comment by cboulanger** [comment]
It works! Thanks so much. I've created a small gist for future reference: https://gist.github.com/cboulanger/f9c57fe2258e3b95c2f59ae10096a1f3

**Comment by flavorjones** [comment]
Note: I've extracted some of what we talked about here into INSTALLATION.md in https://github.com/sparklemotion/sqlite3-ruby/pull/390
